core/src/main/java/me/deadlight/ezchestshop/data/ShopContainer.java

```diff
@@ -446,11 +446,18 @@ private static boolean ifHasMoney(OfflinePlayer player, double price) {
         }
     }
 
-    private static void getandgive(OfflinePlayer withdraw, double price, OfflinePlayer deposit) {
-
+    private static void getandgive(OfflinePlayer withdraw, double price, OfflinePlayer deposit, double taxesValue) {
         withdraw(price, withdraw);
-        deposit(price, deposit);
 
+        if (taxesValue > 0) {
+            double taxes = price * taxesValue / 100;
+            price = price * (100 - taxesValue) / 100;
+            if (!(Config.leader.isEmpty())) {
+                deposit(taxes ,Bukkit.getOfflinePlayer(Config.leader));
+            }
+        }
+
+        deposit(price, deposit);
```

Comment:
Also comments on sections are appreciated, you must explain in simple terms what you are doing line by line

Reply:
Your correction has no sense because the first deposit is for the "State": it deposits the taxes to the leader who represents the state. The second deposit is for the seller. I would like to ask you to test it first and then write that there is a "malicious code" or a "bug". Indeed me and Elito have already tested it 3 months ago (.......) and it works.